Ports NSV map verb

Raven-Industries opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Fixes our map verb and brings functionality from NSV. 
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
requested by franc
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
fix: fixes mapverb
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Raven-Industries committed:
t

Raven-Industries committed:
mapfix

Raven-Industries committed:
Merge remote-tracking branch 'Upstream/master' into mapverb

Raven-Industries committed:
t

KubeRoot commented on _maps/corgstation.json:
```diff
@@ -1,5 +1,6 @@
 {
     "map_name": "Corg Station",
+	"map_link": "CorgStation",
```
Inconsistent indentation, should be spaces

Raven-Industries replied:
whoops, will fix

Raven-Industries commented:
Github is hard.

Raven-Industries committed:
Update corgstation.json

ike709 reviewed (dismissed):
I don't like how this handles maps that don't actually have webmaps but I'm too sleepy to elaborate.

Otherwise based PR

Archanial commented on code/datums/map_config.dm:
```diff
@@ -15,6 +15,7 @@
 
 	// Config actually from the JSON - should default to Box
 	var/map_name = "Box Station"
+	var/map_link = null //This is intentionally wrong, this will make it not link to webmap.
```
box has its map unlike runtime

KubeRoot replied:
The actual link for box is present, in the json for it, I believe this is just a default for when the json doesn't have a map_link or it fails to load entirely.

Raven-Industries replied:
KubeRoot is right, this is just the default value, if Box is the selected map, then this will be updated properly

Archanial replied:
If the box is chosen as a fallback it should still have link to it

Raven-Industries replied:
If I set the default to box, then if, say, Metastation is selected, but Meta doesn't have a map_link defined in the json, then the map verb will provide a link to the box webmap, rather than letting the user know there is no correct map_link

francinum replied:
This is intentional and frankly unavoidable.

KubeRoot commented on interface/interface.dm:
```diff
@@ -256,17 +256,9 @@ Any-Mode: (hotkey doesn't need to be on)
 	set name = "map"
 	set desc = "View the current map in the webviewer"
 	set hidden = 1
```
Actually, why is the verb hidden?

Raven-Industries replied:
I'm not really sure, that's how I found it. Do you want me to change that?

francinum replied:
It's hidden because it was originally tied to a UI element that no longer exists. I see no real reason for it to be hidden.

Raven-Industries committed:
properly sets up maps with no webmap

Raven-Industries committed:
Merge remote-tracking branch 'origin/mapverb' into mapverb

ike709 commented:
Following up, rather than setting link names for maps that don't have webmaps (like multiz debug) just so you don't log an error isn't great. Instead you should just be able to set it to `"None"` which will just null it to give players the error message but not log any errors. Also I highly doubt anybody is going to notice just a `log_world()` pointing out missing maplinks.

Raven-Industries committed:
adds handling for stations without webmaps

Raven-Industries commented:
> Following up, rather than setting link names for maps that don't have webmaps (like multiz debug) just so you don't log an error isn't great. Instead you should just be able to set it to `"None"` which will just null it to give players the error message but not log any errors. Also I highly doubt anybody is going to notice just a `log_world()` pointing out missing maplinks.

@.ike709  Is this what you had in mind? (see latest commit)

francinum commented:
I have no idea if this code suggestion will actually go through, but this is a far better way of doing it.

francinum commented:
```diff
diff --git a/interface/interface.dm b/interface/interface.dm
--- interface/interface.dm
+++ interface/interface.dm
@@ -255,12 +255,13 @@
 /client/verb/map() // i couldn't be fucked to config-ize this
 	set name = "map"
 	set desc = "View the current map in the webviewer"
 	set hidden = 1
-	if(SSmapping.config.map_link && SSmapping.config.map_link !="None")	
+
+	if(SSmapping.config.map_link == "None")
+		to_chat(src,"<span class='danger'>The current map does not have a webmap. </span>")
+	else if(SSmapping.config.map_link)
 		if(alert("This will open the current map in your browser. Are you sure?",,"Yes","No")!="Yes")
 			return
 		src << link("https://affectedarc07.github.io/SS13WebMap/BeeStation/[SSmapping.config.map_link]")
-	else if(SSmapping.config.map_link == "None")
-		to_chat(src,"<span class='danger'>The current map does not have a webmap. </span>")
 	else
 		to_chat(src, "<span class='danger'>The current map is either invalid or unavailable. Open an issue on the github. </span>")

```

Raven-Industries committed:
4th times the charm?

makes if statement less terrible and sets map verb to unhidden (hopefully doesn't break the layout)

KubeRoot commented on interface/interface.dm:
```diff
@@ -255,18 +255,12 @@ Any-Mode: (hotkey doesn't need to be on)
 /client/verb/map() // i couldn't be fucked to config-ize this
```
```suggestion
/client/verb/map()
```

It has been done

Raven-Industries committed:
it has been done

Co-authored-by: KubeRoot <kubinator4321@gmail.com>

francinum commented on interface/interface.dm:
```diff
@@ -252,21 +252,15 @@ Any-Mode: (hotkey doesn't need to be on)
 
-/client/verb/map() // i couldn't be fucked to config-ize this
+/client/verb/map()
 	set name = "map"
```
```suggestion
	set name = "View Webmap"
```

francinum committed:
Update interface/interface.dm
